### PR TITLE
Add conditional elastic_transport import

### PR DESCRIPTION
**Pull request by tammy-baylis-swi**
# Description

Fixes errors from `ElasticsearchInstrumentor().instrument()` when using `elasticsearch` version 8.0.0+. In [version 8](https://www.elastic.co/guide/en/elasticsearch/client/python-api/master/release-notes.html#rn-8-0-0), elasticsearch Python was split into `elasticsearch` and `elastic_transport` packages. This makes `wrapt.wrap_function_wrapper` fail with the current call args. Example trace after `opentelemetry-bootstrap` and `opentelemetry-instrument` with a custom distro when `elasticsearch>=8.0.0` is installed:

```
Instrumenting of elasticsearch failed
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/opentelemetry/instrumentation/auto_instrumentation/sitecustomize.py", line 84, in _load_instrumentors
    distro.load_instrumentor(entry_point, skip_dep_check=True)
  File "/usr/local/lib/python3.7/site-packages/opentelemetry/instrumentation/distro.py", line 63, in load_instrumentor
    instrumentor().instrument(**kwargs)
  File "/usr/local/lib/python3.7/site-packages/opentelemetry/instrumentation/instrumentor.py", line 109, in instrument
    **kwargs
  File "/usr/local/lib/python3.7/site-packages/opentelemetry/instrumentation/elasticsearch/__init__.py", line 150, in _instrument
    response_hook,
  File "/usr/local/lib/python3.7/site-packages/wrapt/wrappers.py", line 895, in wrap_function_wrapper
    return wrap_object(module, name, FunctionWrapper, (wrapper,))
  File "/usr/local/lib/python3.7/site-packages/wrapt/wrappers.py", line 841, in wrap_object
    (parent, attribute, original) = resolve_path(module, name)
  File "/usr/local/lib/python3.7/site-packages/wrapt/wrappers.py", line 829, in resolve_path
    original = lookup_attribute(parent, attribute)
  File "/usr/local/lib/python3.7/site-packages/wrapt/wrappers.py", line 827, in lookup_attribute
    return getattr(parent, attribute)
AttributeError: module 'elasticsearch' has no attribute 'Transport'
Failed to auto initialize opentelemetry
```

The error does not happen when `elasticsearch==7.17.0` is installed.

Fixes https://github.com/open-telemetry/opentelemetry-python-contrib/issues/1809

## Type of change

Please delete options that are not relevant.

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] This change requires a documentation update

# How Has This Been Tested?

1. Set up a Docker container (e.g. `python:3.7`) using local `opentelemetry-python-contrib` as volume
2. `pip install elasticsearch==8.0.0` (repeat later with `7.17.0`)
3. `pip install -Ie /opentelemetry-python-contrib/instrumentation/opentelemetry-instrumentation-elasticsearch`
4. `opentelemetry-bootstrap --action=install`
5. `opentelemetry-instrument <some_service>` where some_service connects to a running es instance and has `OTEL_TRACES_EXPORTER=console`
6. Above error trace no longer prints at service startup and instrumentation.
7. Console traces exporter receives span created by the elasticsearch instrumentation. For example:

```
{
    "name": "Elasticsearch/",
    "context": {
        "trace_id": "0x517dde5598396aab051aef2ad3086dc1",
        "span_id": "0x6f982765b58e1e8a",
        "trace_state": "[]"
    },
    "kind": "SpanKind.CLIENT",
    "parent_id": null,
    "start_time": "2023-05-19T21:23:32.029952Z",
    "end_time": "2023-05-19T21:23:32.103753Z",
    "status": {
        "status_code": "UNSET"
    },
    "attributes": {
        "db.system": "elasticsearch",
        "elasticsearch.url": "/",
        "elasticsearch.method": "GET"
    },
    "events": [],
    "links": [],
    "resource": {
        "attributes": {
            "telemetry.sdk.language": "python",
            "telemetry.sdk.name": "opentelemetry",
            "telemetry.sdk.version": "1.18.0.dev0",
            "service.name": "my-flask-service"
        },
        "schema_url": ""
    }
}
```

I've added setup for running the existing tests with es8 in [935caa6](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/1810/commits/935caa629fd52c8fbac267dca36ef894acbdaf36). They are commented out for now until this related issue is resolved: https://github.com/open-telemetry/opentelemetry-python-contrib/issues/620

# Does This PR Require a Core Repo Change?

- [ ] Yes. - Link to PR: 
- [X] No.

# Checklist:

See [contributing.md](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/main/CONTRIBUTING.md) for styleguide, changelog guidelines, and more.

- [ ] Followed the style guidelines of this project
- [x] Changelogs have been updated
- [ ] Unit tests have been added
- [ ] Documentation has been updated

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,11 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## Unreleased
 
+### Fixed
```
Comment (ocelotl) [inline CHANGELOG.md:10]:
```suggestion
```

File: instrumentation/opentelemetry-instrumentation-elasticsearch/tests/helpers_es8.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
+from elasticsearch_dsl import (  # pylint: disable=unused-import
```
Comment (ocelotl) [inline instrumentation/opentelemetry-instrumentation-elasticsearch/tests/helpers_es8.py:1]: Missin license, please also add a comment that explains why importing `Keyword` is necessary here.

File: instrumentation/opentelemetry-instrumentation-elasticsearch/tests/test_elasticsearch.py [hunk 3]
```diff
@@ -37,7 +37,9 @@
 
 major_version = elasticsearch.VERSION[0]
```
Comment (ocelotl) [inline instrumentation/opentelemetry-instrumentation-elasticsearch/tests/test_elasticsearch.py:38]: Just curious, [here](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/1810/files#diff-ded129d963536138c625950c76c22f4879af17beafb49943c796af9848eea72bR103) you get the version from another attribute.

**Review by ocelotl — APPROVED** [review]

File: instrumentation/opentelemetry-instrumentation-elasticsearch/tests/test_elasticsearch.py (hunk 3 above)
Comment (tammy-baylis-swi) [reply instrumentation/opentelemetry-instrumentation-elasticsearch/tests/test_elasticsearch.py:38]: Ah! Changed the src code check in [775bf2b](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/1810/commits/775bf2b22e567c2dbca7ec60f600dbd955e86253) to match the existing test.

**Comment by ocelotl** [comment]
Please fix lint: https://github.com/open-telemetry/opentelemetry-python/blob/main/CONTRIBUTING.md#development

**Comment by tammy-baylis-swi** [comment]
> Please fix lint: https://github.com/open-telemetry/opentelemetry-python/blob/main/CONTRIBUTING.md#development

Thanks! Addressed in [81432fc](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/1810/commits/81432fc981f252fbf5f5441d445334a6e83e04a2) and lint passes now: https://github.com/open-telemetry/opentelemetry-python-contrib/actions/runs/5204816716/jobs/9389521509?pr=1810

Some parts of `test_celery_functional` are failing, `test_fn_exception_expected` and `test_class_task_exception_excepted`: https://github.com/open-telemetry/opentelemetry-python-contrib/actions/runs/5204816716/jobs/9389521191 I don't think this is related to the changes in this PR.

**Comment by tammy-baylis-swi** [comment]
Thanks @shalevr for merging in main! Is anything else needed for this PR or could it be merged?

No more changelog conflicts and all checks now passing.

**Review by shalevr — APPROVED** [review]

**Pull request merged**